econml/iv/dml/_dml.py

```diff
@@ -527,192 +983,56 @@ def fit(self, Y, T, Z, X=None, *, sample_weight=None, freq_weight=None, sample_v
-
-    @property
-    def nuisance_scores_Y_X(self):
-        """
-        Get the scores for Y_X model on the out-of-sample training data
-        """
-        return self.nuisance_scores_[0]
-
     @property
-    def nuisance_scores_T_X(self):
-        """
-        Get the scores for T_X model on the out-of-sample training data
-        """
-        return self.nuisance_scores_[1]
+    def bias_part_of_coef(self):
+        return self.ortho_learner_model_final_._model_final._fit_cate_intercept
 
     @property
-    def nuisance_scores_T_XZ(self):
-        """
-        Get the scores for T_XZ model on the out-of-sample training data
-        """
-        return self.nuisance_scores_[2]
-
-    def cate_feature_names(self, feature_names=None):
-        """
-        Get the output feature names.
+    def fit_cate_intercept_(self):
+        return self.ortho_learner_model_final_._model_final._fit_cate_intercept
 
-        Parameters
-        ----------
-        feature_names: list of strings of length X.shape[1] or None
-            The names of the input features. If None and X is a dataframe, it defaults to the column names
-            from the dataframe.
+    def shap_values(self, X, *, feature_names=None, treatment_names=None, output_names=None, background_samples=100):
+        if hasattr(self, "featurizer_") and self.featurizer_ is not None:
+            X = self.featurizer_.transform(X)
+        feature_names = self.cate_feature_names(feature_names)
+        return _shap_explain_joint_linear_model_cate(self.model_final_, X, self._d_t, self._d_y,
+                                                     self.bias_part_of_coef,
+                                                     feature_names=feature_names, treatment_names=treatment_names,
+                                                     output_names=output_names,
+                                                     input_names=self._input_names,
+                                                     background_samples=background_samples)
 
-        Returns
-        -------
-        out_feature_names: list of strings or None
-            The names of the output features :math:`\\phi(X)`, i.e. the features with respect to which the
-            final constant marginal CATE model is linear. It is the names of the features that are associated
-            with each entry of the :meth:`coef_` parameter. Not available when the featurizer is not None and
-            does not have a method: `get_feature_names(feature_names)`. Otherwise None is returned.
-        """
-        if feature_names is None:
-            feature_names = self._input_names["feature_names"]
-        if self.original_featurizer is None:
-            return feature_names
-        return get_feature_names_or_default(self.original_featurizer, feature_names)
+    shap_values.__doc__ = LinearCateEstimator.shap_values.__doc__
 
 
-class DMLIV(LinearModelFinalCateEstimatorMixin, _BaseDMLIV):
+class LinearDMLIV(DMLIV):
```

Comment:
As we discussed in sdk meeting several weeks ago, it doesn't support inference

Reply:
Since we are not using the inference capabilities of the final stage OLS, is there really any need for this class? We could just make the default final model in DML be LinearRegression(fit_intercept=False) and then not support final stage inference. The extra class was more there to support final inference. 

Also it seems that maybe we should enable the "summary" capabilities and all the other capabilities that are inherited from the LinearModelFinalCateEstimatorMixin, (except the inference part).

Reply:
Make sense, I will remove LinearDMLIV and only keep DMLIV and make default as linear regression.

Reply:
Summary capabilities have still not been enabled for DMLIV, even though it has a linear final model. We could not implement that if complicated. Just good to have.

Reply:
I exposed the `coef_` and `intercept_` properties, since summary also could only return `point_estimate`, I think `coef_` and `intercept_` is enough?

Reply:
summary is always much more useful than coef_ and intercept, because it has the associated featurizedd names with it and much easier to read. I very rarely find myself using coef_ and intercept_ I almost always use summary.

Reply:
okay, no problem! I will add `summary` here too!